### ISSUE TITLE
[iOS] Configurable currentTimeTimer interval

### ISSUE THREAD

**Issue by wkrause13** [florent37/Flutter-AssetsAudioPlayer#305]
I have a use case that requires sub second progress updates. This doesn't seem to be a problem on Android, but iOS updates occur every second for currentPosition. I assume this was done as a sensible default for performance reasons. I have identified the code in `darwin/Classes/Music.swift` that governs this behavior. 

```swift
        self.currentTimeTimer = Timer.scheduledTimer(timeInterval: 1, target: self, selector: #selector(updateTimer), userInfo: nil, repeats: true)
```

Would you be open to me submitting a pull request that made this value configurable or was there a reason that the 1 second interval was chosen that would prevent such an option? Manually changing it locally seems to work fine.

**Comment by florent37** [comment]
you can update it, but the next method `updateTimer` do this : 

```
@objc func updateTimer(){
        //log("updateTimer")
        if let p = self.player {
            if let currentItem = p.currentItem {
                self.currentTime = CMTimeGetSeconds(currentItem.currentTime())
            }
        }
    }
```

And can only retrieve seconds from the audio : `CMTimeGetSeconds` 

If you find a way to retrieve the current timer as MilliSeconds it would be awesome !!!! but I did not find a way :'(

**Comment by AtixD** [comment]
@florent37 Hey, 
Perhaps, That link can help https://github.com/luanpotter/audioplayers/blob/master/darwin/Classes/AudioplayersPlugin.m#L590

**Comment by florent37** [comment]
I can try yes

**Comment by florent37** [comment]
I implemented it, and set it at 0.3s (300ms), you want to have a custom timer ? or 300ms is good ?

**Comment by florent37** [comment]
```
        self.currentTimeTimer = Timer.scheduledTimer(timeInterval: 0.3, target: self, selector: #selector(updateTimer), userInfo: nil, repeats: true)
```

**Comment by florent37** [comment]
if no I have to add it as argument from flutter to native :'(

**Comment by florent37** [comment]
the 0.3 interval (& milliseconds transformation) has been published in `2.0.9+2`

please try & if you want a fully customizable interval, please reopen this issue :)

**Comment by wkrause13** [comment]
Wow that was fast!

My use case actually requires a slightly more frequent interval. My app is syncing some animations with individual words as they are read, so 300ms kind of works, but something closer to 50ms looks a bit better. 

I know where to adjust that in the code, so no worries if you feel like making the interval configurable only for mac/iOS is something you're not looking to support through the API. Thanks for making this great library and for looking at this!
